src/pagination/index.tsx

```diff
@@ -84,105 +114,158 @@ export default factory(function Pagination({ middleware: { theme, icache, i18n }
 
 	if (pageSizes !== undefined && pageSizes !== icache.get('pageSizes')) {
 		icache.set('pageSizes', pageSizes);
-		icache.set('pageSizesData', pageSizes.map((ps) => ({ value: ps.toString() })));
+		icache.set(
+			'pageSizesResource',
+			createMemoryResourceWithData(pageSizes.map((ps) => ({ value: ps.toString() })))
+		);
+	}
+
+	if (theme.get() !== icache.get('theme')) {
```

Comment:
I think you should be able to use `diffProperty` here for this.
Could you try:

```ts
diffProperty(‘theme’, (current, next) => {
     if (current.theme !== next.theme) { 
      // recalculate here?
    }
});
```

Reply:
Good call. I should use that more.